mongomock/aggregate.py

```diff
@@ -488,6 +489,35 @@ def _handle_type_convertion_operator(self, operator, values):
                 'You need to import the pymongo library to support decimal128 type.'
             )
 
+        # Document: https://docs.mongodb.com/manual/reference/operator/aggregation/toDecimal/
+        if operator == '$toDecimal':
+            if not decimal_support:
+                raise NotImplementedError(
+                    'You need to import the pymongo library to support decimal128 type.'
+                )
+            parsed = self.parse(values)
+            decimal_value = None
+            if isinstance(parsed, bool):
+                parsed = '0' if parsed is True else '1'
+                decimal_value = decimal128.Decimal128(parsed)
+            elif isinstance(parsed, float) or isinstance(parsed, int):
+                decimal_value = decimal128.Decimal128(str(parsed))
+            elif isinstance(parsed, decimal128.Decimal128):
+                decimal_value = parsed
+            elif isinstance(parsed, str):
+                if parsed.isdigit():
+                    decimal_value = decimal128.Decimal128(parsed)
+                else:
+                    raise OperationFailure(
```

Comment:
Please add a test for `"-23"` or `"1.55"`, those should fail, then please fix it.

Reply:
You are right, I have fixed it and add UTs